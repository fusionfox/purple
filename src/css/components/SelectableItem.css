.SelectableItem-item {
  margin-right: 10px;
}
