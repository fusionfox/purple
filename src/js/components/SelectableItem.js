import React from 'react'
import '../../css/components/SelectableItem.css'

const SelectableItem = (props) => {
  return (
    <label className='SelectableItem-label'>
      <input
        className='SelectableItem-item'
        type='checkbox'
        name='product'
        value={props.id}
        onChange={props.handleItemSelection}
      />
      {props.name}
    </label>
  )
}

export default SelectableItem
