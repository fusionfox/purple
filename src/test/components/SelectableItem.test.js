import React from 'react'
import renderer from 'react-test-renderer'
import SelectableItem from '../../js/components/SelectableItem'

describe('SelectableItem component', () => {
  it('Renders a checkbox input with a label', () => {
    const component = renderer.create(
      <SelectableItem
        name='Product Name'
        id='product1'
        handleItemSelection={() => null}
      />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
