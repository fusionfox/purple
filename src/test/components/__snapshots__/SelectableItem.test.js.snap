// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectableItem component Renders a checkbox input with a label 1`] = `
<label
  className="SelectableItem-label"
>
  <input
    checked={false}
    className="SelectableItem-item"
    name="product"
    onChange={[Function]}
    type="checkbox"
    value="product1"
  />
  Product Name
</label>
`;
